saleor/plugins/openid_connect/plugin.py

```diff
@@ -0,0 +1,253 @@
+            scope += " offline_access"
+        return OAuth2Session(
+            client_id=self.config.client_id,
+            client_secret=self.config.client_secret,
+            scope=scope,
+        )
+
+    def external_authentication(
+        self, data: dict, request: WSGIRequest, previous_value
+    ) -> dict:
+        storefront_redirect_url = data.get("redirectUrl")
+        validate_storefront_redirect_url(storefront_redirect_url)
+        uri, state = self.oauth.create_authorization_url(
+            self.config.authorization_url,
+            redirect_uri=prepare_redirect_url(self.PLUGIN_ID, storefront_redirect_url),
+        )
+        return {"authorizationUrl": uri}
+
+    def external_refresh(
+        self, data: dict, request: WSGIRequest, previous_value
+    ) -> dict:
+        refresh_token = request.COOKIES.get(JWT_REFRESH_TOKEN_COOKIE_NAME, None)
+        refresh_token = data.get("refreshToken") or refresh_token
+
+        validate_refresh_token(refresh_token, data)
+        saleor_refresh_token = jwt_decode(refresh_token)  # type: ignore
+        token_endpoint = self.config.token_url
+        try:
+            token_data = self.oauth.refresh_token(
+                token_endpoint,
+                refresh_token=saleor_refresh_token["oauth_refresh_token"],
+            )
+        except AuthlibBaseError:
+            logger.warning("Unable to refresh the token.", exc_info=True)
+            raise ValidationError(
+                {
+                    "refresh_token": ValidationError(
+                        "Unable to refresh the token.",
+                        code=PluginErrorCode.INVALID.value,
+                    )
+                }
+            )
+        try:
+            parsed_id_token = get_parsed_id_token(
+                token_data, self.config.json_web_key_set_url
+            )
+            user = get_user_from_token(parsed_id_token)
+            return get_valid_auth_tokens_from_auth_payload(
+                token_data, user, parsed_id_token
+            )
+        except AuthenticationError as e:
+            raise ValidationError(
+                {
+                    "refreshToken": ValidationError(
+                        str(e), code=PluginErrorCode.INVALID.value
+                    )
+                }
+            )
+
+    def authenticate_user(self, request: WSGIRequest, previous_value) -> Optional[User]:
+        # TODO this will be covered by tests and modified after we add Auth backend for
+        #  plugins
+        token = get_token_from_request(request)
+        if not token:
+            return None
+        return get_user_from_access_token(token)
+
+    @convert_error_to_http_response
+    def handle_auth_callback(self, request: WSGIRequest) -> HttpResponse:
+        storefront_redirect_url = request.GET.get("redirectUrl")
+        if not storefront_redirect_url:
+            raise AuthenticationError("Missing get param - redirectUrl")
```

Comment:
👮‍♂️   Doing my grammar police job
```suggestion
            raise AuthenticationError("Missing GET parameter - redirectUrl")
```

Reply:
😞

Reply:
@maarcingebala You didn't point the missing dot here 😄

Reply:
😢